app/controllers/concerns/hyku_addons/files_controller_behavior.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+module HykuAddons
+  module FilesControllerBehavior
+    extend ActiveSupport::Concern
+
+    # /api/v1/tenant/:tenant_id/files/:file_set_id/work
+    def work
+      render plain: file_set.parent.attributes.dig("id"), status: 200
+    end
+
+    private
+
+      def file_set
+        @file_set ||= begin
+          return unless file_presenter&.current_ability&.can?(:read, file_document.id)
+
+          FileSet.find(file_document.id)
+        end
+      end
```

Comment:
It looks like file set presenter has a parent method which will hit solr instead of fedora so this could be reduced to:
```suggestion
      render plain: file_presenter.parent&.id, status: 200
    end

    private
```

Reply:
Thanks @cjcolvar thats great.